lms/djangoapps/instructor/views/api.py

```diff
@@ -2143,7 +2144,54 @@ def rescore_problem(request, course_id):
     else:
         return HttpResponseBadRequest()
 
-    response_payload['task'] = 'created'
+    response_payload['task'] = TASK_SUBMISSION_OK
+    return JsonResponse(response_payload)
+
+
+@transaction.non_atomic_requests
+@require_POST
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('instructor')
+@require_post_params(problem_to_reset="problem urlname to reset", score='overriding score')
+@common_exceptions_400
+def override_problem_score(request, course_id):
+    course_key = CourseKey.from_string(course_id)
+    score = strip_if_string(request.POST.get('score'))
+    problem_to_reset = strip_if_string(request.POST.get('problem_to_reset'))
+    student_identifier = request.POST.get('unique_student_identifier', None)
```

Comment:
So perhaps this is just me going into paranoia mode, or just misunderstanding what security guarantees are being made here (please let me know if this check happens later). I agree that the decorator means that the caller is an instructor for this course. The line below that reads:

```python
usage_key = UsageKey.from_string(problem_to_reset).map_into_course(course_key)
```

Is going to make sure that the usage key is going to refer to something in the course (as opposed to a usage key in another course). But the only reason `map_into_course()` is used is because we have support old style keys that do not have course key information already attached to the `UsageKey`. When we remove old mongo courses (which is a long term goal), we're likely to remove all calls of `map_into_course` throughout the codebase. We don't want to get into a situation where I am a CCX coach and I can also change my score for arbitrary usage keys.

Anyhow, I'd like to see some access check that this instructor has permission for that piece of content, but I'm ok with just making it a comment above the `map_into_course()` explaining that it can't be trivially removed because of the security implications.